tests/regression/test_ensembleparallelism.py

```diff
@@ -187,6 +187,18 @@ def test_ensemble_reduce(ensemble, mesh, W, urank, urank_sum, root, blocking):
     else:
         assert errornorm(Function(W).assign(10), u_reduce) < 1e-4
 
+    # check that u_reduce dat vector is still synchronised
+    ensemble_rank = ensemble.ensemble_comm.rank
+    spatial_rank = ensemble.comm.rank
+    if ensemble_rank == root:
+        from numpy import zeros
```

Comment:
So here we are just looking at ensemble 0. Which may have multiple processes in it.